lahja/endpoint.py

```diff
@@ -133,6 +170,23 @@ def event_loop(self) -> asyncio.AbstractEventLoop:
 
         return self._loop
 
+    def check_event_loop(func: TFunc) -> TFunc:  # type: ignore
+        """
+        All Endpoint methods must be called from the same event loop.
+        """
+        @functools.wraps(func)
```

Comment:
I think this decorator might accidentally remove type information from any method that it is applied on. There is a `mypy` setting that would catch this that we currently don't have enabled for this project.

```
--disallow-any-decorated  Disallow functions that have Any in their
                            signature after decorator transformation
```

Reply:
I double checked this and it turns out, that this decorator does in fact preserve the type information. Also, we **do run** mypy in strict mode in this project. So, carry on :)

Reply:
Heh, it's cool that `functools.wraps` does the right thing. I think that even if it didn't the outer signature (`def check_event_loop(func: TFunc) -> TFunc`) tells mypy that we're not changing the signature of any of the functions this wraps.